pkg/reconcilermanager/controllers/rootsync_controller.go

```diff
@@ -102,33 +103,54 @@ func (r *RootSyncReconciler) Reconcile(ctx context.Context, req controllerruntim
 	defer r.lock.Unlock()
 
 	rsRef := req.NamespacedName
-	log := r.log.WithValues("rootsync", rsRef.String())
 	start := time.Now()
 	var err error
 	rs := &v1beta1.RootSync{}
 	reconcilerRef := types.NamespacedName{
 		Namespace: configmanagement.ControllerNamespace,
 		Name:      core.RootReconcilerName(rsRef.Name),
 	}
+	log := r.log.WithValues(
+		logFieldObject, rsRef.String(),
+		logFieldKind, r.syncKind,
+		logFieldReconciler, reconcilerRef.String())
 
 	if err = r.client.Get(ctx, rsRef, rs); err != nil {
 		metrics.RecordReconcileDuration(ctx, metrics.StatusTagKey(err), start)
 		if apierrors.IsNotFound(err) {
 			r.clearLastReconciled(rsRef)
-			r.log.Info("Deleting managed objects",
-				logFieldObject, rsRef.String(),
-				logFieldKind, r.syncKind)
-			return controllerruntime.Result{}, r.deleteClusterRoleBinding(ctx, reconcilerRef)
+			// TODO: Handle cleanup of RootSync deleted while reconciler-manager is down.
+			if !r.hasCachedSync(rsRef) {
+				log.Error(err, "Sync object not managed by this reconciler-manager",
+					logFieldObject, rsRef.String(),
+					logFieldKind, r.syncKind)
+				// return `controllerruntime.Result{}, nil` here to make sure the request will not be requeued.
+				return controllerruntime.Result{}, nil
+			}
+			// Cleanup after already deleted RootSync.
+			// This code path is unlikely, because the custom finalizer should
+			// have already deleted the managed resources and removed the
+			// rootSyncs cache entry. But if we get here, clean up anyway.
+			if err := r.deleteManagedObjects(ctx, reconcilerRef, nil, rs, false); err != nil {
```

Comment:
Need to add a check to see if the `configsync.gke.io/reconciler` is present. If so, log that the reconciler-manager finalizer is blocked waiting on the reconciler finalizer. 

Otherwise, the reconciler will be deleted before the user objects have been deleted.